samples/buildstrategy/kaniko/buildstrategy_kaniko-trivy_cr.yaml

```diff
@@ -0,0 +1,94 @@
+# This Build Strategy will intentionally fail if the image has any
+# critical CVEs. It will not be pushed into the destination registry
+# if any critical vulnerabilities are found.
+---
+apiVersion: shipwright.io/v1alpha1
+kind: ClusterBuildStrategy
+metadata:
+  name: kaniko-trivy
+spec:
+  buildSteps:
+    - name: kaniko-build
+      image: gcr.io/kaniko-project/executor:v1.6.0
+      workingDir: $(params.shp-source-root)
+      securityContext:
+        runAsUser: 0
+        capabilities:
+          add:
+            - CHOWN
+            - DAC_OVERRIDE
+            - FOWNER
+            - SETGID
+            - SETUID
+            - SETFCAP
+            - KILL
+      env:
+        - name: HOME
+          value: /tekton/home
+        - name: AWS_ACCESS_KEY_ID
+          value: NOT_SET
+        - name: AWS_SECRET_KEY
+          value: NOT_SET
+      command:
+        - /kaniko/executor
+      args:
+        - --skip-tls-verify=true
+        - --dockerfile=$(build.dockerfile)
+        - --context=$(params.shp-source-context)
+        - --destination=$(params.shp-output-image)
+        - --oci-layout-path=/kaniko/oci-image-layout
+        - --snapshotMode=redo
+        - --no-push
+        - --tarPath
+        - /tar/image.tar
+      volumeMounts:
+        - name: tar
+          mountPath: /tar/
+      resources:
+        limits:
+          cpu: 500m
+          memory: 1Gi
+        requests:
+          cpu: 250m
+          memory: 65Mi
+    # Reference:
+    # https://ibm-gsi-ecosystem.github.io/ibm-gsi-cloudnative-journey/developer-advanced-2/devsecops-trivy/
+    - name: trivy-scan
+      image: docker.io/aquasec/trivy:0.19.2
+      volumeMounts:
+      - mountPath: /image/
+        name: tar
+      command:
+        - trivy
+      args:
+        - --exit-code=1
+        - --severity=CRITICAL
+        - --input
+        - /image/image.tar
+      resources:
+        limits:
+          cpu: 500m
+          memory: 1Gi
+        requests:
+          cpu: 250m
+          memory: 65Mi
+
+    # Reference:
+    # https://ibm-gsi-ecosystem.github.io/ibm-gsi-cloudnative-journey/developer-advanced-2/devsecops-trivy/
```

Comment:
You may leave this out of the yaml, you could include it in the commit message or PR text.